FIX #29901 supporting non-invoiceable task time in invoice generation process

vdegrandpre opened the pull request:
REPLACED BY #29964 

Fixes an issue where the user add non-invoiceable time to an invoice by generating it from the time spent on project tasks list.
At the same time, add a safeguard to disable the select checkbox in two conditions : task is not invoiceable ou task time already invoiced. The checkbox can be re-enabled by double-clicking very close outside the desired checkbox to enable iut again.

eldy committed:
Doc

FHenry committed:
fix: #29864, do not allow more than 1 serial in mass stock transfert (#29898)

atm-lena committed:
New llx header body css product stats/stock (#29894)

* llxHeader : bodycsss product - stats part

* llxHeader : bodycsss product - stock part

---------

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

eldy committed:
Fix HTML5

Thomas905 committed:
fix (#29873)

nicolas-eoxia committed:
Fix CSS class not used and deprecated in HTML 5 (#29880)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

Daoud-mohamed committed:
fix warning (#29904)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

Quentin-Seekness committed:
Add price (and multicurrency price) in arrayfields (#29869)

FHenry committed:
fix: fix SQL new install (#29861)

* fix: SQL fresh install script

* fix: SQL fresh install script

* fix perms files

* fix: travis CI

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

Commit:
NEW : llxHeader body css for knowledgemanagement #27152  (#29867)

* Work on issue 26856 for knowledgemanagement

* Work on issue 26856 for knowledgemanagement

* Work on issue 26856 for knowledgemanagement

* Update htdocs/knowledgemanagement/knowledgerecord_list.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

lamrani002 committed:
Fix type variable PHPStan (#29893)

* Fix type variable PHPStan

* fix phan

* Update modulebuilder.lib.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

atm-maxime committed:
NEW : add indexes on chargesociales table (#29911)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

FHenry committed:
fix: cookie JS creation missing defaut attribute SameSite (#29862)

* fix: cookie JS creation missing defaut setting

* fix: travis CI

* fix: travis CI

* fix: travis CI

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

vdegrandpre committed:
supporting non-invoiceable task time in invoice generation process

vdegrandpre committed:
pre-commit

vdegrandpre committed:
Add langs from timesheet

vdegrandpre committed:
oups pré-commit

vdegrandpre committed:
Modification de terme

vdegrandpre committed:
phan - retreait réf. timesheet

vdegrandpre committed:
Protection quand il n'y a pas d'extrafields

vdegrandpre committed:
parenthèse du tabernacle

frederic34 committed:
fix inline images with swiftmailer (#29923)

defrance committed:
FIX : create mrp table if enabled (#29921)

* create mrp table if enabled

* Update modMrp.class.php

frederic34 committed:
remove deprecated handler no more needed (#29919)

atm-lena committed:
llxHeader : bodycsss product -  others part (#29918)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

EchoLoGeek committed:
Add llxHeader body css for resource (#29924)

Hystepik committed:
Fix email colllector collection to add date in meta data description (#29920)

* Fix email colllector collection to add date in meta data description

* Update emailcollector.class.php

---------

Co-authored-by: Hystepik <lmarcouiller@nltechno.com>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy commented on htdocs/projet/tasks/time.php:
```diff
@@ -1566,13 +1566,22 @@ function setDetailVisibility() {
-		$sql .= " t.invoice_line_id";
+		$sql .= " t.invoice_line_id,";
+		if ($invoiceable) {
+			$sql .= " efpt.invoiceable as options_invoiceable";
```
This field does not exist by default in dolibarr so should not be used in default code.

vmaury committed:
Clean expedition list code (#29868)

* Clean expedition list code (replace $shiment by $object in list table where both were used)

* Clean expedition list code (replace $shiment by $object in list table where both were used)

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

iouston committed:
remove warning phpstan (#29866)

Property ActionComm::$libelle is unused.
Is it deliberate to keep this property depreciated?

atm-jpb committed:
New user author modele fac rec (#29858)

* add conf  in fac_rec

* pre-commit

* Update facture.class.php

Signed-off-by: Laurent Destailleur <eldy@destailleur.fr>

* Update facture.class.php

Signed-off-by: Laurent Destailleur <eldy@destailleur.fr>

---------

Signed-off-by: Laurent Destailleur <eldy@destailleur.fr>
Co-authored-by: jpb <jean-pascal.boudet@atm-consulting>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

EchoLoGeek committed:
Add llxHeader to tickets (#29932)

Hystepik committed:
Fix dol_escape_htmltag to prevent bad strip of char (#29931)

Co-authored-by: Hystepik <lmarcouiller@nltechno.com>

atm-lena committed:
llxHeader CSSBody - admin part 1 (#29930)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

frederic34 committed:
fix phpstan Expression "'@phan-var string[] $prim%.... (#29928)

frederic34 committed:
Update mails.lang fix pre-commit (#29927)

atm-lena committed:
llxHeader CSSBody - user (#29926)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

lamrani002 committed:
Fix delete linked files for ticket and Don (#29929)

frederic34 committed:
fix an old error (#29850)

* fix an old error

* fix OdfException is never thrown in the try block

* fix OdfException is never thrown in the try block

* fix Undefined array key "name"

* fix missing translation

* fix missing translation

* fix Undefined variable $lineid

* fix warnings

* fix upload dir

* fix phpcs

atm-lena committed:
New llx header body css admin part2 (#29933)

* llxHeader CSSBody - admin part 2

* clean

* Fix param llxHeader

---------

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

FHenry committed:
fix: #29909 (#29910)

* fix: #29909

* reveiw

EchoLoGeek committed:
Add llxHeader to reception (#29935)

eldy committed:
Look and feel v20

creekorful committed:
Allow tweaking of the index that MySQL use when searching products (#29934)

* Allow tweaking of the index that MySQL use when searching products

Sometimes when working with large dataset with many entities the MySQL engine does not pick-up the right index.
This is especially true when doing a search over the reference because the LIKE '%%' prevent the uk_product_label from being chosen.

This PR aims to improve this situation by doing the following :

- Adding a simple index on entity column.
- Adding an hidden constant to force the index being used when searching products.

Performances over a 8M products dataset shows a reduction of 2/3 execution time.

* Add missing global $conf

* Fix wrong usage of getDolGlobalString()

* Finalise fix

* Add more indexes

* Rework indexes

eldy committed:
FIX #29915 Better fix to show full path of category in multiselect mode.

Daoud-mohamed committed:
Fix warnings technical debt (#29907)

* fix warning

* Technical debt

* fix

rycks committed:
prefix and suffix core cleanup (#29938)

rycks committed:
Fix #29640:  (#29939)

* FIX REPLENISH MANY FOURN WHEN ORDER ALREADY CREATE (#29710)

* fix restrictedArea on selectobject (#29882)

* Fix travis

* Fix : expense report reapproval email, wrong date format + missing parameter (#29874)

* delete_linked_contact is < 0 in case of error

---------

Co-authored-by: IC-Mathieu <95343351+IC-Mathieu@users.noreply.github.com>
Co-authored-by: Rémi Champlon <85104766+Kazimir42@users.noreply.github.com>
Co-authored-by: Laurent Destailleur <eldy@users.sourceforge.net>
Co-authored-by: Maxime Kohlhaas <maxime@atm-consulting.fr>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

FHenry committed:
fix: Mass stock transfert display error message when something goes wrong  (#29905)

* fix: check in MvtStockClass consistancy of serial batch quantity

* fix: check in MvtStockClass consistancy of serial batch quantity

* fix UI

atm-lena committed:
llxHeader - BodyCss - webhook/webservice/workstation (#29953)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

atm-lena committed:
llxHeader BodyCss - export part (#29950)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

atm-lena committed:
llxHeader BodyCss - takepos part (#29952)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

frederic34 committed:
fix phpstan level 4 (#29954)

* fix phpstan level 4

* fix phpstan level 4

zahwyrix committed:
NEW : Execute doActions on statistics pages (#29851)

* NEW : Execute doActions on statistics pages

* Update index.php

* Update index.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Fix responsive

frederic34 committed:
fix phpstan level 4 (#29949)

Dead catch - OdfException is never thrown in the try block. method only throw SegmentException

frederic34 committed:
fix phpstan (#29948)

atm-lena committed:
llxHeader CSSBody - commande (#29947)

* llxHeader CSSBody - commande

* llxHeader BodyCss - delivery part

---------

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

atm-lena committed:
llxHeader CSSBody - barcode (#29946)

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

atm-lena committed:
llxHeader CSSBody - bookmark (#29945)

* llxHeader CSSBody - bookmark

* admin_bookmark edit

---------

Co-authored-by: atm-lena <lena.papazian@atm-consulting.fr>

atm-maxime committed:
FIX : test on button and action must be the same (#29944)

eldy commented:
Right. Adding the field "billable" into the table project_task is a better place...

Hystepik committed:
Fix display of non html messages in ticket/messaging.php (#29896)

* Fix display of non html messages in ticket/messaging.php

* Fix pr

* fix remaining nl2br

* add maildate

* fix conflict

---------

Co-authored-by: Hystepik <lmarcouiller@nltechno.com>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

atm-florianm committed:
NEW convenience methods for getting / setting the value of an extrafield (#29856)

* NEW convenience methods for getting / setting the value of an extrafield on an object

* Update htdocs/core/class/commonobject.class.php

[code formatting] removing extra space

vdegrandpre committed:
intégration billable à l'obet tâche

vdegrandpre committed:
billable project task attribute management

vdegrandpre committed:
pre-commit
